### ISSUE TITLE
[om/next] :advanced compilation warnings in 1.0.0pre31

### ISSUE THREAD

**Issue by danskarda** [omcljs/om#643]
After upgrade to new pre31, I get warnings with :advanced optimization mode. There are two warnings for each defui in a file. Maybe new version references some symbol without React externs? The warning was not present in pre30.

```
WARNING: Use of undeclared Var foo.bar.baz/x28103 at line 64 /home/user/work/project/src/cljs/foo/bar/baz.cljs
WARNING: Use of undeclared Var foo.bar.baz/x28103 at line 64 /home/user/work/project/src/cljs/foo/bar/baz.cljs
WARNING: Use of undeclared Var foo.bar.baz/x28137 at line 84 /home/user/work/project/src/cljs/foo/bar/baz.cljs
WARNING: Use of undeclared Var foo.bar.baz/x28137 at line 84 /home/user/work/project/src/cljs/foo/bar/baz.cljs
```

**Comment by currentoor** [comment]
I'm getting the same warning in `1.0.0-alpha32`.

**Comment by egracer** [comment]
Same error for me in 1.0.0-alpha37. Maybe the macro is confusing advanced compilation with respect to defining externs?

**Comment by anmonteiro** [comment]
This is just a bogus warning that's awaiting a fix upstream.
See http://dev.clojure.org/jira/browse/CLJS-1607 for details

**Comment by anmonteiro** [comment]
This has just been [fixed upstream](https://github.com/clojure/clojurescript/commit/273bcd33522b085fe98948598528fcf2c90a778c) and will be in the next version of ClojureScript

**Comment by anmonteiro** [comment]
Solved in Clojurescript as of version 1.9.183
